spec/person_spec.rb

```diff
@@ -0,0 +1,24 @@
+require_relative '../person'
+
+describe Person do
+  context 'When passing a name' do
+    person = Person.new(id: nil, age: 25, name: 'sadiq', parent_permission: true)
+    person_two = Person.new(id: nil, age: 12, name: 'dire', parent_permission: false)
+    it 'should return first letter capital' do
+      expect(person.validate_name).to eq 'Sadiq'
+    end
+
+    it 'should return true if he can use service' do
+      expect(person.can_use_services?).to eq true
+    end
+
+    it 'should return false if he cannot use service' do
+      expect(person_two.can_use_services?).to eq false
+    end
+
+    it 'should add into rentals' do
+      person.add_rentals(person)
+      expect(person.rentals.length).to eq 1
+    end
+  end
+end
```

Comment:
- [ ] Let us add a few tests to cover all the functionalities for some instance methods in the Person file.

`it "The add_rental method returns a rental and adds its to the book"`
`it "The constructor method sets a default value for the optional name parameter if it is not provided`
`it "The constructor method returns a Person object with provided age, name, and empty rentals"`